File: src/slang/parse/ParseException.java
package slang.parse;

import slang.lex.Token;

public class ParseException extends RuntimeException {
    public final Token token;

    ParseException(Token token, String reason) {
        super(reason);
        this.token = token;
    }

    @Override
    public String getMessage() {
        return String.format("[line %d, col %d] ", token.line, token.col) + super.getMessage();
    }
}

File: src/slang/parse/ParseException.scala
package slang.parse

import slang.lex.Token

class ParseException(val token: Token, val reason: String) extends RuntimeException(reason) {
    override def getMessage(): String = {
        String.format("[line %d, col %d] ", token.loc.line, token.loc.col) + super.getMessage();
    }
}
